parsers/OPENNEM.py

```diff
@@ -203,8 +203,11 @@ def fetch_main_power_df(
     # Solar rooftop is a special case
     df = process_solar_rooftop(df)
     logger.debug("Preparing capacities..")
-    capacities = get_capacities(filtered_datasets, region)
-    return df, capacities
+    if region:
+        capacities = get_capacities(filtered_datasets, region)
+        return df, capacities
+    else:
+        return df
```

Comment:
At first glance this looks confusing - how would the function's callers know if it will return 1 or 2 things? Having looked at it I see that `fetch_exchange` calls `df = fetch_main_power_df(...)` while `fetch_production` calls `df, capacities = fetch_main_power_df(...)`.

I would suggest to have this function always return a tuple:

```suggestion
    if region:
        capacities = get_capacities(filtered_datasets, region)
        return df, capacities
    else:
        return df, None
```

function return value type annotation: `Tuple[pd.DataFrame, Optional[pd.Series]]`

change the call in `fetch_exchange` to throw away the second value: `df, _ = fetch_main_power_df(...)`

Reply:
Thanks for the feedback, and I definitely agree that it's a bit confusing right now :D 
I tried improving it a bit without going too deep, what do you think?